### ISSUE TITLE
SparkeyPairSCollection#asSparkey does not create files for empty SCollection

### ISSUE THREAD

**Issue by sisidra** [spotify/scio#1723]
If `SCollection[String, String]` is empty then calling `asSparkey` will not result in any error and will not create an empty Sparkey file.
If that is expected behaviour, can it be documented in scaladocs?

**Comment by ClaireMcGinty** [comment]
As someone who has never personally used Sparkey, I think writing an empty file seems like a reasonable default behavior and would match the behavior of other sinks (text, avro, and BQ I/Os will all create empty files).  Seems like this could easily be achieved by moving `SparkeyWriter`'s instantiation & closing outside of the Transform itself. @jto , @andrisnoko , do either of you have strong opinions on this...?

**Comment by sisidra** [comment]
I agree that empty file should be expected behavior.
Moving SparkeyWriter outside should not work tho as those handles will be accessed on different machines :)
Something like this might work:
```scala
in
  .map(Some(_))
  .union(sc.parallelize[Option[(String, String)]](None))
  .groupBy(...)
  .map(
    // ...
    val it = xs.iterator.flatten
    // ...
  )
```
https://github.com/spotify/scio/blob/master/scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/package.scala#L127-L140

Maybe I can prepare PR.

**Comment by ClaireMcGinty** [comment]
ah good point, I was assuming some different behavior of the SparkeyWriter! looking at your PR now.

**Comment by nevillelyh** [comment]
Fixed in #1834
